main/src/main/java/com/ldtteam/blockout/compat/UpdateHandler.java

```diff
@@ -0,0 +1,96 @@
+package com.ldtteam.blockout.compat;
+
+import com.ldtteam.blockout.connector.server.ServerGuiController;
+import com.ldtteam.blockout.gui.BlockOutGuiData;
+import com.ldtteam.blockout.inventory.BlockOutContainerData;
+import com.ldtteam.blockout.inventory.BlockOutContainerLogic;
+import com.ldtteam.blockout.management.server.update.ServerUpdateManager;
+import com.ldtteam.blockout.proxy.ProxyHolder;
+import com.ldtteam.blockout.util.Log;
+import com.ldtteam.jvoxelizer.IGameEngine;
+import com.ldtteam.jvoxelizer.client.gui.IGuiContainer;
+import com.ldtteam.jvoxelizer.common.gameevent.event.ITickEvent;
+import com.ldtteam.jvoxelizer.common.gameevent.event.player.IPlayerGameEvent;
+import com.ldtteam.jvoxelizer.entity.living.player.IMultiplayerPlayerEntity;
+import com.ldtteam.jvoxelizer.inventory.IContainer;
+import com.ldtteam.jvoxelizer.util.distribution.executor.IDistributionExecutor;
+import org.apache.commons.lang3.time.StopWatch;
+
+import java.util.concurrent.TimeUnit;
+
+public class UpdateHandler
+{
+    private static Thread updateThread = null;
+
+    public static void onPlayerLoggedOut(final IPlayerGameEvent.ILoggedOutEvent event)
+    {
+        if (event.getPlayerEntity() instanceof IMultiplayerPlayerEntity)
+        {
+            IMultiplayerPlayerEntity playerMP = (IMultiplayerPlayerEntity) event.getPlayerEntity();
+
+            ProxyHolder.getInstance().getGuiController().closeUI(playerMP);
+        }
+    }
+
+    public static void onTickClientTick(final ITickEvent.IClientTickEvent event)
+    {
+        if (event.getPhase() != ITickEvent.Phase.END)
+        {
+            return;
+        }
+
+        IDistributionExecutor.onClient(() -> {
+            if (IGameEngine.getInstance().getCurrentGui() instanceof IGuiContainer && IGameEngine.getInstance().getCurrentGui().getInstanceData() instanceof BlockOutGuiData)
+            {
+                IGuiContainer<BlockOutGuiData> currentScreen = (IGuiContainer<BlockOutGuiData>) IGameEngine.getInstance().getCurrentGui();
+                currentScreen.getInstanceData().getRoot().getUiManager().getUpdateManager().updateElement(currentScreen.getInstanceData().getRoot());
+
+                ClientTickManager.getInstance().onClientTick();
+            }
+        });
+    }
+
+    public static void onTickServerTick(final ITickEvent.IServerTickEvent event)
+    {
+        IDistributionExecutor.onServer(() -> {
```

Comment:
Is this supposed to not run on integrated server?

Reply:
It is supposed to run both on the Integrated server as well as the dedicated server.
Maybe i should rename IDistribution / IDistributionExecutor to ILogicalSide / ILogicalExecutor.

Reply:
Ah so that's what was confusing to me... forge 1.13 considers Dist to be dedeicaded server/client, not logical side.